amazons3/amazons3_endpoint.bal

```diff
@@ -87,26 +86,32 @@ public remote function Client.getBucketList() returns Bucket[]|error {
 
     request.setHeader(HOST, self.amazonHost);
     request.setHeader(X_AMZ_CONTENT_SHA256, UNSIGNED_PAYLOAD);
-    generateSignature(request, self.accessKeyId, self.secretAccessKey, self.region, GET, requestURI, UNSIGNED_PAYLOAD);
-
-    var httpResponse = self.amazonS3Client->get("/", message = request);
-    if (httpResponse is http:Response) {
-        int statusCode = httpResponse.statusCode;
-        var amazonResponse = httpResponse.getXmlPayload();
-        if (amazonResponse is xml) {
-            if (statusCode == 200) {
-                return getBucketsList(amazonResponse);
-            } else {
+    var signature = generateSignature(request, self.accessKeyId, self.secretAccessKey, self.region, GET, requestURI,
+        UNSIGNED_PAYLOAD);
+
+    if (signature is error) {
+        error err = error(AMAZONS3_ERROR_CODE, { cause: signature,
+            message: "Error occurred while generating the amazon signature header" });
+        return err;
+    } else {
+        var httpResponse = self.amazonS3Client->get("/", message = request);
+        if (httpResponse is http:Response) {
+            int statusCode = httpResponse.statusCode;
+            var amazonResponse = httpResponse.getXmlPayload();
+            if (amazonResponse is xml) {
+                if (statusCode == 200) {
+                    return getBucketsList(amazonResponse);
+                }
                 return setResponseError(statusCode, amazonResponse);
+            } else {
+                error err = error(AMAZONS3_ERROR_CODE, { message: "Error occurred while accessing the xml payload
+                            of the response." });
```

Comment:
This will add unnecessary space between "payload" and "of".

Shall we change to something like
```ballerina
                error err = error(AMAZONS3_ERROR_CODE, 
                                  { message: "Error occurred while accessing the xml payload of the response." });
```